### PR TITLE
Replace non-standard __progname with argv[0]

### PR DESCRIPTION
**Pull request by andreblanke**

### CODE REVIEW

**Comment by jcs** [comment]
What platform are you on that doesn't define `__progname`?

**Comment by zvezdochiot** [comment]
https://stackoverflow.com/questions/273691/using-progname-instead-of-argv0

File: xbanish.c [hunk 1]
```diff
@@ -444,10 +442,10 @@ snoop_legacy(Window win)
 }
 
 void
-usage(void)
+usage(char *progname)
```
Comment (zvezdochiot) [inline xbanish.c]: Lift to top of file?

Comment (andreblanke) [reply xbanish.c]: I don't quite understand why we'd need to lift it to the top of the file. For stylistic reasons?

**Comment by andreblanke** [comment]
@jcs As linked by @zvezdochiot certain combinations of OSX/clang.
Also, the fact that a standard-conformant alternative exists which is usable (which would not be the case if it were a library, as far as I know) and barely any more difficult than `__progname` warrants this change for me.
Why deviate from the standard if one doesn't necessarily have to?

File: xbanish.c (hunk 1 above)
Comment (zvezdochiot) [reply xbanish.c]: Yes. And this eliminates the unnecessary definition of the ```usage``` function.

**Comment by jcs** [comment]
Referencing an 11-year old post on Stack Overflow isn't really helping your case.

>  Why deviate from the standard if one doesn't necessarily have to?

If `argv[0]` was the same as `__progname`, no one would have deviated from this "standard", but they have.  That's because `__progname` is not the same as `argv[0]`.  Here's the output before:

```
$ /usr/local/bin/xbanish -h
xbanish: unknown option -- h
usage: xbanish [-a] [-d] [-i mod] [-m nw|ne|sw|se]
```

And here it is with your change:

```
$ /usr/local/bin/xbanish -h
/usr/local/bin/xbanish: invalid option -- 'h'
usage: /usr/local/bin/xbanish [-a] [-d] [-i mod] [-m nw|ne|sw|se]
```

Now it's showing the full path name when that is not what is desired. If you want to make this change, you also need to also call `basename()` on `argv[0]` before printing it.

**Comment by zvezdochiot** [comment]
@jcs say:
>you also need to also call basename() on argv[0] before printing it.

What for?

**Comment by jcs** [comment]
Because that's how I want it to look, that's why I used `__progname` in the first place and not `argv[0]`.

**Comment by zvezdochiot** [comment]
@jcs say:
>Because that's how I want

Close PR? Not?

**Comment by jcs** [comment]
This isn't your pull request, why do you even care?

**Comment by zvezdochiot** [comment]
@jcs say:
>why do you even care?

KISS.

**Comment by andreblanke** [comment]
> Now it's showing the full path name when that is not what is desired. If you want to make this change, you also need to also call basename() on argv[0] before printing it.

Then feel free to close this PR if it does not fit your vision of the program. I just stopped by and saw it as an opportunity to remove something non-standard.
I don't see anything wrong with printing the full path name instead of just the executable name but it is your program after all.

**Pull request closed without merging**